### ISSUE TITLE
docker/runc doesn't work on Fedora 26

### ISSUE THREAD

**Issue by robertgzr** [kinvolk/kubeadm-nspawn#45]
Full kubelet error:
<details>

```
kubeadm-nspawn-0 kubelet[82]: E0622 17:23:33.029071      82 pod_workers.go:182] Error syncing pod 7075157cfd4524dbe0951e00a8e3129e ("etcd-kubeadm-nspawn-0_kube-system(7075157cfd4524dbe0951e00a8e3129e)"), skipping: failed to "CreatePodSandbox" for "etcd-kubeadm-nspawn-0_kube-system(7075157cfd4524dbe0951e00a8e3129e)" with CreatePodSandboxError: "CreatePodSandbox for pod \"etcd-kubeadm-nspawn-0_kube-system(7075157cfd4524dbe0951e00a8e3129e)\" failed: rpc error: code = 2 desc = failed to start sandbox container for pod \"etcd-kubeadm-nspawn-0\": Error response from daemon: {\"message\":\"invalid header field value \\\"oci runtime error: container_linux.go:247: starting container process caused \\\\\\\"process_linux.go:359: container init caused \\\\\\\\\\\\\\\"rootfs_linux.go:53: mounting \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"cgroup\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\" to rootfs \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"/var/lib/docker/overlay/c211c41b8e46e1cdd7def0298d675db90eb563a6fac0d17b7d32db71264ea979/merged\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\" at \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"/sys/fs/cgroup\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\" caused \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"no subsystem for mount\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\"\\\\\\\"\\\\n\\\"\"}"
```
</details>

---

Reproduce: `docker run --rm -it busybox` inside coreos container spawned by `kubeadm-nspawn` on Fedora 26
```
docker: Error response from daemon: invalid header field value "oci runtime error: container_linux.go:247: starting container process caused \"process_linux.go:359: container init caused \\\"rootfs_linux.go:53: mounting \\\\\\\"cgroup\\\\\\\" to rootfs \\\\\\\"/var/lib/docker/overlay/550589b3e61fe73d1e5fdba641f562796e056c3ba92e877a04562dc34fe8a237/merged\\\\\\\" at \\\\\\\"/sys/fs/cgroup\\\\\\\" caused \\\\\\\"no subsystem for mount\\\\\\\"\\\"\"\n".
```

inside the coreos container:
- docker v1.12.6
- systemd v233

Possible ways to fix:

- change cgroup-driver of kubelet/docker to **cgroupfs** and upgrade runc to at least `1.0rc3` (default docker on coreos uses runc `1.0rc2`)
- turn off off unified cgroup hierarchy with `systemd.legacy_systemd_cgroup_controller=1`

**Comment by alban** [comment]
Could we specify `systemd.legacy_systemd_cgroup_controller=1` in the Vagrantfile?

**Comment by robertgzr** [comment]
The docker command above works on the fedora host with docker v1.13.1 and systemd v233

**Comment by robertgzr** [comment]
Can be fixed by switching to CoreOS Alpha `v1451.2.0` [see](https://coreos.com/releases/#1451.2.0)

**Comment by robertgzr** [comment]
Fixed via #32
